avocado/utils/nvme.py

```
@@ -304,19 +463,118 @@ def create_full_capacity_ns(controller_name, shared_ns=False):
+    # This skips the FLBAS=0 attempt and goes straight to optimal detection
+    if flbas == -1:
+        try:
+            flbas = get_optimal_flbas(controller_name)
+            LOGGER.info(
+                f"Auto-detected FLBAS={flbas} for namespace creation on {controller_name}"
+            )
+        except NvmeException as e:
+            LOGGER.error(f"Failed to auto-detect FLBAS: {e}")
+            raise NvmeException(
+                f"Cannot create namespace: FLBAS auto-detection failed: {e}"
+            )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)